Intel SoC FPGA Agilex Development Kit Initial Bring Up

siclim opened the pull request:
This is the initial Zephyr support for Intel SoC FPGA Agilex support.
Agilex has quad-core 64-bit Arm Cortex*-A53.

This patch build Zephyr for Agilex development kit with 256KB SDRAM and
support hello_world sample code. The Zephyr will need to be loaded by
Intel Arm Trusted Firmware (ATF).

Agilex Zephyr boot flow:
	FSBL:ATF BL2(EL3) -> SSBL:ATF BL31(EL3) -> OS:Zephyr(EL2->EL1)

Intel ATF can be downloaded from:
	https://github.com/altera-opensource/arm-trusted-firmware.git

siclim committed:
soc: arm64: Add MMU table for Intel SoC FPGA Agilex

Add MMU table for Intel SoC FPGA Agilex initial bring up.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

siclim committed:
dts: arm64: Add device tree for Intel SoC FPGA Agilex

Add device tree for Intel SoC FPGA Agilex initial bring up. This is the
first version of device tree which enable single CPU core and basic
drivers that needed by hello_world example.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

siclim committed:
boards: arm64: Build Zephyr for Intel SoC FPGA Agilex development kit

This is the initial Zephyr support for Intel SoC FPGA Agilex support.
Agilex has quad-core 64-bit Arm Cortex*-A53.

This patch build Zephyr for Agilex development kit with 256KB SDRAM and
support hello_world sample code. The Zephyr will need to be loaded by
Intel Arm Trusted Firmware (ATF).

Agilex Zephyr boot flow:
	FSBL:ATF BL2(EL3) -> SSBL:ATF BL31(EL3) -> OS:Zephyr(EL2->EL1)

Intel ATF can be loaded from:
	https://github.com/altera-opensource/arm-trusted-firmware.git

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

siclim committed:
drivers: clock_control: Add clock driver for Intel SoC FPGA Agilex

Add clock driver for Intel SoC FPGA Agilex.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

siclim committed:
soc: arm64: Add common header files for Intel SoC FPGA

Add common header files for Intel SoC FPGA. The header files define the
handoff data structure and system manager base addresses needed by
Agilex initial bring up.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

siclim committed:
soc: arm64: Add Kconfig files for Intel SoC FPGA

Add Kconfig files for intel_socfpga folder.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

carlocaione commented on boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig:
```diff
@@ -0,0 +1,32 @@
+CONFIG_ARM_ARCH_TIMER=y
+
+# Zephyr Kernel Configuration
+CONFIG_XIP=n
+CONFIG_FLASH_SIZE=0
+CONFIG_FLASH_BASE_ADDRESS=0x0
+CONFIG_SRAM_BASE_ADDRESS=0x10000000
+CONFIG_SRAM_SIZE=256
+CONFIG_AARCH64_IMAGE_HEADER=y
```
None of this should actually be needed.
- `CONFIG_XIP` is `n` by default
- `CONFIG_AARCH64_IMAGE_HEADER` is `y` by default now, 
- `CONFIG_SRAM_*` should be defaulted to whatever is in the DTS
- same for `CONFIG_FLASH_*`

siclim replied:
Noted. I will remove this when submit version 2 for review.

carlocaione commented on drivers/clock_control/CMakeLists.txt:
```diff
@@ -40,3 +40,8 @@ else()
 endif()
+
+if(CONFIG_SOC_SERIES_AGILEX)
+  zephyr_sources(clock_agilex_ll.c)
+  zephyr_sources(clock_agilex.c)
```
Why not using `zephyr_sources_ifdef`?

siclim replied:
Noted. I will update to

zephyr_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex_ll.c)
zephyr_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex.c)

carlocaione commented on drivers/clock_control/clock_agilex_ll.c:
```diff
@@ -0,0 +1,392 @@
+		data = mmio_read_32(CLKMGR_OFFSET + CLKMGR_STAT);
+		count++;
+		if (count >= 1000)
+			return -ETIMEDOUT;
```
you are returning `uint32_t`

siclim replied:
I will update the function to return 'int'. Same for the next function.

dcpleung replied:
A single line statement for if-block needs curly brackets. Please add.

carlocaione commented on drivers/clock_control/clock_agilex_ll.c:
```diff
@@ -0,0 +1,392 @@
+		data = mmio_read_32(CLKMGR_OFFSET + CLKMGR_STAT);
+		count++;
+		if (count >= 1000)
+			return -ETIMEDOUT;
```
same

carlocaione commented on drivers/clock_control/clock_agilex_ll.c:
```diff
@@ -0,0 +1,392 @@
+		count++;
+	} while ((req_status & CLKMGR_MEM_REQ) && (count < 10));
+
+	if (count >= 100)
```
This `if` is actually never taken because `count` is at most `10` given the while loop two lines before.

siclim replied:
Thanks. I will fix the code to to (count >= 10).

carlocaione commented on soc/arm64/intel_socfpga/agilex/mmu_regions.c:
```diff
@@ -0,0 +1,36 @@
+			      0x08E00000,
+			      MT_DEVICE_nGnRnE | MT_RW | MT_SECURE),
+
+	MMU_REGION_FLAT_ENTRY("MEMORY",
+			      DT_REG_ADDR(DT_NODELABEL(mem0)),
+			      DT_REG_SIZE(DT_NODELABEL(mem0)),
+			      MT_NORMAL | MT_RW | MT_SECURE),
```
Mapping the whole memory should not be necessary.

siclim replied:
Thanks. I have removed this.

carlocaione commented on drivers/clock_control/clock_agilex_ll.c:
```diff
@@ -0,0 +1,392 @@
+	return rdata;
+}
+
+void config_clkmgr_handoff(struct handoff *hoff_ptr)
```
What is using this function?

siclim replied:
Hi Carlo,
We plan to support and upstream Agilex features by phases.

There will be 2 kinds of Zephyr boot for Agilex:

(1) ATF BL2 -> ATF BL31 -> Zephyr
     - This is what we upstream now, first batch of of Agilex upstream code

(2) Zephyr self-boot, processor directly boot from zephyr.bin
     - We will upstream this flow later. 
     - We will need to call 'z_arm64_el3_plat_init' function to initialize hardware component (example: pinmux, clock)
     - config_clkmgr_handoff will be used later once we enable self-boot flow. And, the clock_agilex_ll.c is shared/migrated from our ATF source code. Thus, we retains the function for future use.

Kindly advise this is acceptable for the mainline code.  Thanks a lot.

carlocaione replied:
If you are upstreaming now only the ATF boot path I would expect only to find in this PR code related to that path. So, please remove any unused / not needed code for this PR, you can add that later when upstreaming the other boot flow.

siclim replied:
Noted, I will remove all unused code for this phase.

carlocaione commented on soc/arm64/intel_socfpga/agilex/mmu_regions.c:
```diff
@@ -0,0 +1,36 @@
+	MMU_REGION_FLAT_ENTRY("PERIPHERAL",
+			      0xF7000000,
+			      0x08E00000,
+			      MT_DEVICE_nGnRnE | MT_RW | MT_SECURE),
```
Use `MT_P_RW_U_RW` or similar instead of `MT_RW`

siclim replied:
Noted.

carlocaione commented on soc/arm64/intel_socfpga/agilex/mmu_regions.c:
```diff
@@ -0,0 +1,36 @@
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 0),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_SECURE),
```
Are you sure you want this to be `MT_SECURE`?

siclim replied:
I have changed it to MT_NS, it doesn't need to be secure.

carlocaione replied:
Not sure about your use-case but you can think to use `MT_DEFAULT_SECURE_STATE`

siclim replied:
Noted, I will change to use MT_DEFAULT_SECURE_STATE. And, will adjust accordingly in future when needed.

carlocaione commented on soc/arm64/intel_socfpga/agilex/soc.h:
```diff
@@ -0,0 +1,41 @@
+#define SOCFPGA_SOC2FPGA_SCR_REG_BASE           0xffd21200
+#define SOCFPGA_LWSOC2FPGA_SCR_REG_BASE         0xffd21300
+
+#define mmio_write_32(addr, data) \
+	sys_write32((data), (addr))
+
+#define mmio_read_32(addr) \
+	sys_read32((addr))
```
Why do you want to define new macros?

siclim replied:
Hi Carlo,

We will re-use our ATF driver code for Zephyr.  The ATF driver codes are using mmio_X function to write/read. 
We add these macros to map mmio_X to Zephyr compatible function, so that we can re-use most of the code from ATF.

carlocaione replied:
That's not actually correct. You are defining these macros in a soc-specific header file, but you are using the macros in a general driver. Now, let's say (I know this is really unlikely) that someone wants to use your clock agilex driver, in that case they cannot do that (or at least the driver fails to compile) because they have no access to the macros because these are defined in a SoC header file.

So, you have two ways of fixing this:
1. Move the macros out of this header file in a more accessible place so that it can be used together with the drivers using the macros
2. Remove the macros entirely

siclim replied:
Is it acceptable that I move the mmio macro to the beginning of 'clock_agilex_ll.c' itself? 
This allow us to re-use the existing code.

carlocaione replied:
Yes, but please add a comment explaining why, otherwise it looks odd :)

siclim replied:
Hi Carlo,

Noted and thanks.

A side question, I am new to open source upstream process, is that I should close this pull-request now and resubmit a new pull-request after I clean up all codes?

carlocaione replied:
Nope, you have to force push your branch with the changes using git `git push --force ....`, that will retrigger the process.

Please also note that besides addressing my comments you have two other issues to address: the compliance and the scancode checks are failing (see the red crosses at the bottom of this PR).

siclim replied:
Noted and thanks for reminder for the compliance and scancode error. I will check that.

carlocaione commented on soc/arm64/intel_socfpga/agilex/soc.h:
```diff
@@ -0,0 +1,41 @@
+#define mmio_read_32(addr) \
+	sys_read32((addr))
+
+#define mmio_setbits_32(addr, bit) \
+	sys_write32(sys_read32((addr)) | (bit), (addr))
+
+#define mmio_clrbits_32(addr, bit) \
+	sys_write32(sys_read32((addr)) & ~(bit), (addr))
```
Use `sys_set_bit` and `sys_clear_bit` instead.

siclim replied:
Hi Carlo,

In our use case, the 'bit' represent mask, our code will set/clear more that 1 bits using this command. 
I will change bit to mask in the macro to confusion.
    mmio_setbits_32(addr, bit) => mmio_setbits_32(addr, mask)

sys_set_bit and sys_clear_bit inline function accept bit offset. It does not fit our function. To centralize bitops in zephyr, is it possible to provide the macro that set/clear bits using mask?
    Example: sys_set_bits (addr, mask)

static ALWAYS_INLINE void sys_set_bit(mem_addr_t addr, unsigned int bit)
{
	uint32_t temp = *(volatile uint32_t *)addr;

	*(volatile uint32_t *)addr = temp | (1 << bit);
}

Thanks,
Siew Chin

carlocaione replied:
> is it possible to provide the macro that set/clear bits using mask?

Feel free to add the new macros you need in `include/arch/common/sys_bitops.h`

siclim replied:
Noted.

carlocaione commented on soc/arm64/intel_socfpga/common/socfpga_handoff.h:
```diff
@@ -0,0 +1,97 @@
+	uint32_t	_pad_0x618_0x620[2];
+};
+
+int verify_handoff_image(struct handoff *hoff_ptr, struct handoff *reverse_hoff_ptr);
+int socfpga_get_handoff(struct handoff *hoff_ptr);
```
Where are this functions defined?

siclim replied:
These 2 functions are only needed in self-boot flow in socfpga_handoff.c, which I don't upstream in this batch. Sorry for confusion. I will remove these 2 lines.

(2) Zephyr self-boot, processor directly boot from zephyr.bin

carlocaione commented:
Board documentation is missing as well.

siclim committed:
codeowners: Add code owners for Intel SoC FPGA Agilex

Add code owners for Intel SoC FPGA Agilex board and clock drivers.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

siclim committed:
include: common: Add sys_set_bits and set_clear_bits inline functions

Add new common inline functions sys_set_bits and set_clear_bits to set
and clear multiple bits via bit mask in single function call.

Signed-off-by: Siew Chin Lim <elly.siew.chin.lim@intel.com>

dcpleung commented on drivers/clock_control/clock_agilex_ll.c:
```diff
@@ -0,0 +1,392 @@
+		count++;
+	} while ((req_status & CLKMGR_MEM_REQ) && (count < 10));
+
+	if (count >= 100)
+		return -ETIMEDOUT;
```
Same as the comment above about count not reaching 100.

siclim replied:
Noted. I will update the code in next pull request. Thanks.

siclim commented:
Hi Reviewers,

Can you please help to review the pull request?

carlocaione commented on boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts:
```diff
@@ -0,0 +1,28 @@
+
+/dts-v1/;
+
+#include <intel_socfpga/intel_socfpga_agilex.dtsi>
+/ {
```
```suggestion
#include <intel_socfpga/intel_socfpga_agilex.dtsi>

/ {
```

siclim replied:
Okay. Updated.

carlocaione commented on boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts:
```diff
@@ -0,0 +1,28 @@
+
+};
+
+&uart0{
```
```suggestion
&uart0 {
```

siclim replied:
Okay. Updated.

carlocaione commented on drivers/clock_control/clock_agilex.c:
```diff
@@ -0,0 +1,58 @@
+	.get_rate = clk_get_rate
+};
+
+DEVICE_DT_DEFINE(DT_NODELABEL(clock), clk_init, device_pm_control_nop,
```
`device_pm_control_nop` is deprecated, use `NULL`

siclim replied:
Noted, updated.

carlocaione commented on dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi:
```diff
@@ -0,0 +1,73 @@
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+
+/ {
+	cpus{
```
```suggestion
	cpus {
```

siclim replied:
Okay, update.

carlocaione commented on dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi:
```diff
@@ -0,0 +1,73 @@
+		};
+	};
+
+	soc{
```
```suggestion
	soc {
```

siclim replied:
Okay, updated.

carlocaione commented on soc/arm64/intel_socfpga/agilex/mmu_regions.c:
```diff
@@ -0,0 +1,31 @@
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("PERIPHERAL",
+			      0xF7000000,
+			      0x08E00000,
```
What's this huge hard-coded area?

siclim replied:
This is the peripheral register map range for all peripheral in Agilex. We enable for OS access for initial bring.

carlocaione replied:
Uhm, I see two problems here:
1. You are missing at least two device regions here: UART and CLOCK that have entries in the DT (you only have GIC).
2. Why do you need to map the whole peripheral range here? Do you have some kind of application running that needs that?

siclim replied:
UART and CLOCK register are within this huge peripheral register range, we just enable for all during bring up without specific application.

Agreed that this is a problem.

I will do a clean up to only enable for UART and CLOCK, which means only enable the peripheral that we enabled in the code.

siclim replied:
Remove huge area of PERIPERAL from MMU table, only include:
(1) SYSTEM_MANAGER and CLOCK register map for clock driver
(2) UART0 register map for uart driver

siclim commented:
I have updated the patches based on review and force pushed to the branch. Kindly help to review. Thanks a lot.

siclim commented on soc/arm64/intel_socfpga/agilex/mmu_regions.c:
```diff
@@ -0,0 +1,42 @@
+static const struct arm_mmu_region mmu_regions[] = {
+
+	/* System manager register that required by clock driver */
+	MMU_REGION_FLAT_ENTRY("SYSTEM_MANAGER",
+			      0xFFD12000,
+			      0x00001000,
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_DEFAULT_SECURE_STATE),
```
syscon-like device? I am not sure how syscon-like device is, so i can't confirm.

Agilex System Manager register:
https://www.intel.com/content/www/us/en/programmable/hps/agilex/hps.html#jwd1561687644789.html

System Manager is a block inside HPS (Hard processer system) in Agilex SoCFPGA, it provide some settings for the periperal interface within HPS. 

Clock driver will need to read the external clock freq from boot_scratch_cold1 and boot_scratch_cold2 registers in system manager clock. cold1 and cold2 register is defined for custom used, in our implementation, we  use these 2 register to stores the external system clock system from user settings. So we need to access it in clock driver to calculate the frequently.

carlocaione replied:
**syscon** comes from the linux world https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/mfd/syscon.yaml and it represents  _a register region containing a set of miscellaneous registers. The registers are not cohesive enough to represent as any specific type of device_ and unfortunately this is exactly what it looks like to me.

Now, the problem is that you are defining here a device region without a backing device or driver and that should be always avoided but on the other hand Zephyr is not shipping any syscon driver for that.

I cannot ask you to write a syscon driver (even though it would be really easy to do) but please, at least use a couple of defines for the reg / size values and add a comment explaining what this region is.

siclim replied:
I have taka a look at syscon.yaml, and, yes that this is what I need. The system manager register will be access by different driver or feature in future, we just need a way to access the register directly.

I agree with you that it is cleaner to create a "syscon" device tree node for these register range. I can add this new device tree node element and driver. If there any concern if I proceed to add it?

Can we use back "syscon" as the name for this element? I expect this element only need to support 2 properties: "compatible" and "reg". We can add more properties in future when needed.

siclim replied:
I will proceed to add a driver call system_mgr for our device.

carlocaione replied:
I'm writing a syscon driver right now, I will push it tomorrow.

siclim replied:
Noted, thanks. I will wait for your change. :)

cfriedt commented:
@siclim - please resolve conflicts and re-request reviews

siclim commented:
Hi, I have dependency on https://github.com/zephyrproject-rtos/zephyr/pull/35903 and I am still waiting for it to be merged into main.

cfriedt commented:
> Hi, I have dependency on #35903 and I am still waiting for it to be merged into main.

@siclim - you should be good to go now.

siclim commented:
Noted and thanks.

siclim commented:
Sorry for my late update. Can you please help to remove the 'stale' lable and help to review the update?

I have updated the to use 'syscon' device as per request to specify the register map for SYSTEM MANAGER in Agilex. And, MMU will get the start address and size from the sysmgr device tree.

	MMU_REGION_FLAT_ENTRY("SYSTEM_MANAGER",
			      DT_REG_ADDR(DT_NODELABEL(sysmgr)),
			      DT_REG_SIZE(DT_NODELABEL(sysmgr)),
			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_DEFAULT_SECURE_STATE),

nashif commented on include/arch/common/sys_bitops.h:
```diff
@@ -42,6 +42,20 @@ static ALWAYS_INLINE int sys_test_bit(mem_addr_t addr, unsigned int bit)
 	return temp & (1 << bit);
 }
 
+static ALWAYS_INLINE void sys_set_bits(mem_addr_t addr, unsigned int mask)
```
please document the new APIs in include/sys/sys_io.h

siclim replied:
I have added intel_socfpga_agilex_socdk.yaml and run twister but there are "Cmake build failure". Is it expected?

swiplab@PG15SWIPLAB1115:~/siclim/zephyrproject/zephyr$ ./scripts/twister -p intel_socfpga_agilex_socdk
ZEPHYR_BASE unset, using "/home/swiplab/siclim/zephyrproject/zephyr"
Install the anytree module to use the --test-tree option
Install tabulate python module with pip to use --device-testing option.
Install pyserial python module with pip to use --device-testing option.
Install tabulate python module with pip to use --device-testing option.
Install psutil python module with pip to run in Qemu.
can't import pykwalify; won't validate YAML (No module named 'pykwalify')
INFO    - Zephyr version: v2.7.0-rc2-144-g82c1c10d85d7
INFO    - JOBS: 12
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 1466 test scenarios (1406 configurations) selected, 1017 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    3/ 389   0%  skipped: 1020, failed:    0ERROR   - Cmake build failure: /home/swiplab/siclim/zephyrproject/zephyr/samples/modules/nanopb for intel_socfpga_agilex_socdk

ERROR   - intel_socfpga_agilex_socdk samples/modules/nanopb/sample.modules.nanopb       FAILED: Cmake build failure
ERROR   - see: /home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/samples/modules/nanopb/sample.modules.nanopb/build.log

nashif replied:
Not, this is not expected. you need to install the python dependencies, here pykwalify, look at the getting started guide

siclim replied:
Thanks, it can be built after install all the dependencies. There are 2 build error:

twister.log
----------
2021-10-02 04:36:33,849 - twister - ERROR - intel_socfpga_agilex_socdk tests/posix/fs/portability.posix.fs                [31mFAILED[39m: Build failure
2021-10-02 04:36:33,850 - twister - ERROR - see: [33m/home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/tests/posix/fs/portability.posix.fs/build.log[39m
2021-10-02 04:36:33,925 - twister - ERROR - intel_socfpga_agilex_socdk tests/posix/fs/portability.posix.fs.newlib         [31mFAILED[39m: Build failure
2021-10-02 04:36:33,925 - twister - ERROR - see: [33m/home/swiplab/siclim/zephyrproject/zephyr/twister-out/intel_socfpga_agilex_socdk/tests/posix/fs/portability.posix.fs.newlib/build.log[39m

build.log
----------
-- Found BOARD.dts: /home/swiplab/siclim/zephyrproject/zephyr/boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.dts
intel_socfpga_agilex_socdk.dts.pre.tmp:28.21-39.5: Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property
....
zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/build.make:103: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj' failed
make[2]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj] Error 1
CMakeFiles/Makefile2:2796: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all' failed
make[1]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2


I didn't see any error in the build.log, can you please advise is  warning 'missing or empty reg/ranges property' caused the build failed? I have tried to refer to other dts in Zephyr, I do see any difference on soc/timer when compare with others.

nashif replied:
good, you need to resolve those issues and make sure they are not coming with your change.

siclim replied:
I have fixed all warning caused by my changes. 

The twister is failed with the following error which is not caused by my changes. Am i good to resubmit my patches for review?

zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/build.make:103: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj' failed
make[2]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/fat_fs.c.obj] Error 1
CMakeFiles/Makefile2:2798: recipe for target 'zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all' failed
make[1]: *** [zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/all] Error 2
Makefile:90: recipe for target 'all' failed

siclim replied:
I have resubmitted the patches. All warnings coming from my changes are fixed.

nashif reviewed (changes requested):
You need to add intel_socfpga_agilex_socdk.yaml file into the board directory, this will allow integration with the test runner. Once you have the yaml file, try running:

`twister -p intel_socfpga_agilex_socdk`

and make sure everything builds correctly.

siclim commented:
4 updates:

1. Document newly added sys_set_bits and set_clear_bits API in include/sys/sys_io.h
2. Change zephyr_sources_ifdef to zephyr_library_sources_ifdef in drivers/clock_control/CMakeLists.txt to fix the warning reported in twister.
3. Remove soc/ layer in dts/arm64/intel_socfpga/intel_socfpga_agilex.dtsi file to fix 'Warning (simple_bus_reg): /soc/timer: missing or empty reg/ranges property'
4. Add boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk.yaml file to enable unit test run on intel socfpga.

nashif commented:
@carlocaione can you please give this another look?

carlocaione reviewed (approved):
It looks sane to me but @siclim please remove the `Update:` lines from the commit messages (in theory those should not appear in the commit message after merge but not sure how github handles that).

siclim commented:
Hi @anasnashif , 

All checks have passed, can you please help to review again? Thanks a lot.
